src/view/screens/Settings/ExportCarDialog.tsx

```diff
@@ -18,20 +19,17 @@ export function ExportCarDialog({
   const {_} = useLingui()
   const t = useTheme()
   const {gtMobile} = useBreakpoints()
-  const {currentAccount} = useSession()
   const {getAgent} = useAgent()
 
-  const downloadUrl = React.useMemo(() => {
+  const download = React.useCallback(async () => {
     const agent = getAgent()
-    if (!currentAccount || !agent.session) {
-      return '' // shouldnt ever happen
+    if (!agent.session) {
+      return // shouldnt ever happen
     }
-    // eg: https://bsky.social/xrpc/com.atproto.sync.getRepo?did=did:plc:ewvi7nxzyoun6zhxrhs64oiz
-    const url = new URL(agent.pdsUrl || agent.service)
-    url.pathname = '/xrpc/com.atproto.sync.getRepo'
-    url.searchParams.set('did', agent.session.did)
-    return url.toString()
-  }, [currentAccount, getAgent])
+    const did = agent.session.did
+    const res = await agent.com.atproto.sync.getRepo({did})
```

Comment:
Aside from the issue with opening in mail on Android, this promise can take a while to resolve. For me it takes about 10 seconds or so for the download to complete (my CAR file is about 15 MB)

I think what we should do is

- Keep the old functionality on web. The browser should be handling the download there I think.
- Set the button to disabled with a spinner inside of it once you press download on native, so that the user knows something is going on.

I have some local changes that do all of that, but wondering about this mail issue right now 🙏

Reply:
The main reason for me to change this behavior is that I want to remove the code that builds the download URL. Indeed, with the oauth refactor coming up, that URL won't be as straightforward to build and I don't want to add complex logic in the client, when the building of the url is already handled by the client when making the API call. For this reason, I would prefer if we could keep the same behavior on web.

I agree that having a loading state would be significantly better. I didn't add one because I wanted my changes to be minimal. If you have branch with these changes, can you link it here? Or simply add the commit with the loader on this branch?

Reply:
Ahh makes sense! Yes I can push those up here in a bit, would have the other day but didn't want to push to your branch w/o asking first 😄 will clean it up and push it up here a bit later today!

Reply:
I added a loader.